var/spack/repos/builtin/packages/fftx/package.py

```diff
@@ -0,0 +1,90 @@
+# Copyright 2013-2022 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+import subprocess
```

Comment:
We have API documentation for `Executable` at https://spack.readthedocs.io/en/latest/spack.util.html#module-spack.util.executable with `which` at https://spack.readthedocs.io/en/latest/spack.util.html#spack.util.executable.which.

We currently have 26 packages combining `CudaPackage` and `ROCmPackage`.  Of those, several use `which` to determine how to run their stand-alone tests.  For example, `super-dist` relies on `which` and `PATH`, see https://github.com/spack/spack/blob/b08f147cf3f67a0c25d75f3a65615b903aaa5bae/var/spack/repos/builtin/packages/slate/package.py#L103 .

`Sundials` and `trilinos` use `Executable` to find compiler executables at https://github.com/spack/spack/blob/b08f147cf3f67a0c25d75f3a65615b903aaa5bae/var/spack/repos/builtin/packages/sundials/package.py#L254 and https://github.com/spack/spack/blob/b08f147cf3f67a0c25d75f3a65615b903aaa5bae/var/spack/repos/builtin/packages/trilinos/package.py#L813 , respectively.

Similarly, `vtk-m` uses `Executable` to find `cmake` and their test program at https://github.com/spack/spack/blob/b08f147cf3f67a0c25d75f3a65615b903aaa5bae/var/spack/repos/builtin/packages/vtk-m/package.py#L355 and https://github.com/spack/spack/blob/b08f147cf3f67a0c25d75f3a65615b903aaa5bae/var/spack/repos/builtin/packages/vtk-m/package.py#L361 , respectively.

Reply:
@broderickpt wrote:

> I don’t understand why you’d not want standard python, but be that as it may, I don’t know how you use Executable or which – I presume they are something specific to spack as they don’t appear in the normal python docs.  How/where would I find some simple examples illustrating what I’d need in order to use them?

Reply:
Does https://github.com/spack/spack/pull/28583#discussion_r831508230 help?

Reply:
Using which ('bash') to run script.
Package fftx inherits from both CudaPackage and ROCmPackage